src/rules/color-no-hex.ts

```diff
@@ -0,0 +1,50 @@
+import type { Rule } from 'eslint';
+/**
+ * Regular expression to match hex color values.
+ * Added capturing group to extract the hex value.
+ * Using 'u' flag for Unicode support.
+ */
+const hexColorRegex = /(#\p{Hex_Digit}{3,6})\b/u;
+
+/**
+ * Rule to prevent the use of hex color values.
+ */
+export const colorNoHex: Rule.RuleModule = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Prevent the use of hex color values.',
+      recommended: true,
+      url: 'https://github.com/MetaMask/eslint-plugin-design-tokens/blob/main/docs/rules/color-no-hex.md',
+    },
+    schema: [], // This rule does not take any options
+  },
+  create(context) {
+    return {
+      Literal(node) {
+        if (typeof node.value === 'string') {
+          const matches = node.value.match(hexColorRegex);
+          if (matches) {
+            const hexValue = matches[1] as string; // Type assertion to string
+            context.report({
+              node,
+              message: `'${hexValue}' Hex color values are not allowed. Consider using design tokens instead. For more information, visit: https://github.com/MetaMask/design-tokens or reach out to the design system team #metamask-design-system.`,
+            });
```

Comment:
Open to better messaging here!

Reply:
My only opinion on messaging is keeping it short and direct and this current messaging is just fine!

Reply:
perhaps link this doc to the message?

Reply:
Thanks for the suggestions. It makes sense to direct them to the rule documentation rather than the design-token docs. From the rule documentation they can get to the design token docs. And the url is linked in the meta data which I think is what the ESlint plugin uses to direct devs to the docs here https://github.com/MetaMask/eslint-plugin-design-tokens/blob/499c0e44bcf5c512bf413c8cc887758bb1dc59b8/src/rules/color-no-hex.ts#L18